fmriprep/interfaces/reports.py

```diff
@@ -73,7 +73,7 @@ class SubjectSummaryInputSpec(BaseInterfaceInputSpec):
     t2w = InputMultiPath(File(exists=True), desc='T2w structural images')
     subjects_dir = Directory(desc='FreeSurfer subjects directory')
     subject_id = Str(desc='Subject ID')
-    bold = traits.List(desc='BOLD functional series')
+    bold = InputMultiPath(desc='BOLD functional series')
```

Comment:
I can't link a review, for some reason ! I couldn't have `exists=True` since:
> Because the bold_list it's being passed is a list that can contain sub-lists, requiring File(exists=True) was not working for the sub-lists. Unless you can think of another way to check this!

Reply:
I see. So what are all the valid options? One file, a list of files, or a list of lists of files?

If so, I think we can use the following:

```Python
bold = InputMultiPath(traits.Either(File(exists=True), traits.List(File(exists=True)))
```

Reply:
I was surprised this wasn't caught in testing, but I see why, now-- the bug report must only have one functional run, where for the non-ME data sets we tested there were two runs each.

So, yes, valid options are exactly as you listed. For `traits.List(File(exists=True))`, it will still work if the list it's being passed contains sub-lists? That wasn't my experience, but I'm not sure if it's the combination with `InputMultiPath` that's what's helping, here.

Reply:
Right, so `InputMultiPath` allows a trait to take either a single object or a list of such objects. To the interface it will always be seen as a list.

| Input | View |
|-|-|
| `a` | `[a]` |
|`[a]`|  `[a]` |
|`[a, b]` | `[a, b]` |

By setting value of the trait to `traits.Either(File(exists=True), traits.List(File(exists=True))`, we're saying each element `a` or `b` needs to be either a single file or a list of files.

So, one thing with this nested construct is that we need to be careful about how input shapes will be interpreted. The suggestion I gave above is intended to keep the list as flat as possible, while we could do another one to ensure that it's always two levels deep:

```Python
>>> from nipype.interfaces.base import *
>>> class Spec(BaseInterfaceInputSpec): 
...     option1 = InputMultiPath(InputMultiPath(File(exists=True)))        
...     option2 = InputMultiPath(traits.Either(File(exists=True), traits.List(File(exists=True))))
>>> spec = Spec()
>>> for option in ('tmp/img1.nii', ['tmp/img1.nii'], [['tmp/img1.nii']],
...                ['tmp/img1.nii', 'tmp/img2.nii'],
...                [['tmp/img1.nii', 'tmp/img2.nii']],
...                [['tmp/img1.nii'], ['tmp/img2.nii']],
...                [['tmp/img1.nii'], 'tmp/img2.nii'],
...                [['tmp/img1.nii', 'tmp/img2.nii'], ['tmp/img3.nii', 'tmp/img4.nii']],
...                ):
...    print(repr(option))
...    spec.option1 = option
...    spec.option2 = option
...    print(repr(spec))
...    print()
'tmp/img1.nii'

ignore_exception = False
option1 = [['tmp/img1.nii']]
option2 = ['tmp/img1.nii']


['tmp/img1.nii']

ignore_exception = False
option1 = [['tmp/img1.nii']]
option2 = ['tmp/img1.nii']


[['tmp/img1.nii']]

ignore_exception = False
option1 = [['tmp/img1.nii']]
option2 = [['tmp/img1.nii']]


['tmp/img1.nii', 'tmp/img2.nii']

ignore_exception = False
option1 = [['tmp/img1.nii'], ['tmp/img2.nii']]
option2 = ['tmp/img1.nii', 'tmp/img2.nii']


[['tmp/img1.nii', 'tmp/img2.nii']]

ignore_exception = False
option1 = [['tmp/img1.nii', 'tmp/img2.nii']]
option2 = [['tmp/img1.nii', 'tmp/img2.nii']]


[['tmp/img1.nii'], ['tmp/img2.nii']]

ignore_exception = False
option1 = [['tmp/img1.nii'], ['tmp/img2.nii']]
option2 = [['tmp/img1.nii'], ['tmp/img2.nii']]


[['tmp/img1.nii'], 'tmp/img2.nii']

ignore_exception = False
option1 = [['tmp/img1.nii'], ['tmp/img2.nii']]
option2 = [['tmp/img1.nii'], 'tmp/img2.nii']


[['tmp/img1.nii', 'tmp/img2.nii'], ['tmp/img3.nii', 'tmp/img4.nii']]

ignore_exception = False
option1 = [['tmp/img1.nii', 'tmp/img2.nii'], ['tmp/img3.nii', 'tmp/img4.nii']]
option2 = [['tmp/img1.nii', 'tmp/img2.nii'], ['tmp/img3.nii', 'tmp/img4.nii']]
```

Perhaps we should be a little less abstract, though. What are the specific cases we need to handle, given multi-echo inputs? Can you describe what the input lists would look like, to make sure we can distinguish multiple runs from multiple echoes?

Reply:
A grouping with both multi-echo and single-echo data could look something like [this](https://gist.github.com/emdupre/78c34b91669f22b1144e640d0e7a4166#file-group_echos-py-L4).

Thanks for clarifying, though-- I'll update this PR to your suggestion, now, then.